test/sql/function/list/lambdas/list_comprehension.test

```diff
@@ -0,0 +1,27 @@
+# name: test/sql/function/list/lambdas/list_comprehension.test
+# description: Test different examples that can be found when searching for python list comprehension examples
+# group: [lambdas]
+
+# even squares
+query I
+SELECT list_transform(list_filter([0, 1, 2, 3, 4, 5], x -> x % 2 = 0), y -> y * y)
+----
+[0, 4, 16]
+
+# divisible by 2 and 5
+query I
+SELECT list_filter(list_filter([2, 4, 3, 1, 20, 10, 3, 30], x -> x % 2 == 0), y -> y % 5 == 0)
+----
+[20, 10, 30]
+
+query I
+SELECT list_filter(['apple', 'banana', 'cherry', 'kiwi', 'mango'], fruit -> contains(fruit, 'a'))
+----
+[apple, banana, mango]
+
+query I
+SELECT list_transform([[1, NULL, 2], [3, NULL]], a -> list_filter(a, x -> x IS NOT NULL))
+----
+[[1, 2], [3]]
+
+# NOTE: could not find a way to reproduce nested for-loops, e.g. to transpose a matrix
```

Comment:
I don't think we can transpose a matrix, as our `list_apply` can only modify elements within a list, not move elements around between lists as is needed for transposing a matrix.

Reply:
Yes, I just saw this being done with list comprehensions in python and thought 'neat'! But I also think that it is not possible.